sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobServiceVersionPolicyTest.groovy

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob
+
+import com.azure.core.http.policy.HttpLogDetailLevel
+import com.azure.core.http.policy.HttpLogOptions
+import com.azure.core.test.TestMode
+import com.azure.storage.blob.models.BlobContainerEncryptionScope
+import com.azure.storage.blob.models.BlobRange
+import com.azure.storage.blob.models.EncryptionScope
+import com.azure.storage.blob.specialized.PageBlobClient
+
+class BlobServiceVersionPolicyTest extends APISpec {
+
+    BlobClient bc
+    BlobContainerClient cc
+    EncryptionScope es
+    BlobContainerEncryptionScope ces
+
+    BlobContainerClientBuilder containerBuilder
+
+    def setup() {
+        es = new EncryptionScope().setEncryptionScope("testscope1")
+
+        containerBuilder = new BlobContainerClientBuilder()
+            .endpoint(cc.getBlobContainerUrl().toString())
+            .serviceVersion(BlobServiceVersion.V2019_02_02)
+            .httpClient(getHttpClient())
+            .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+            .credential(primaryCredential)
+
+        if (testMode == TestMode.RECORD) {
+            containerBuilder.addPolicy(interceptorManager.getRecordPolicy())
+        }
+    }
+
+    def "Default encryption scope"() {
+        when:
+        ces = new BlobContainerEncryptionScope()
+            .setDefaultEncryptionScope("testscope2")
+            .setDenyEncryptionScopeOverride(true)
+        BlobContainerClient cpkncesContainer = containerBuilder.blobContainerEncryptionScope(ces).encryptionScope(null)
+            .containerName(generateContainerName()).buildClient()
+        cpkncesContainer.createWithResponse(null, null, null, null)
+
+        then:
+        def e = thrown(IllegalArgumentException)
+        e.getMessage() == "x-ms-default-encryption-scope is not supported for create container in service version 2019-02-02"
+    }
+
+    def "Deny encryption scope override"() {
+        setup:
+        ces = new BlobContainerEncryptionScope()
+            .setDefaultEncryptionScope(null)
+            .setDenyEncryptionScopeOverride(true)
+        BlobContainerClient cpkncesContainer = containerBuilder.blobContainerEncryptionScope(ces)
+            .containerName(generateContainerName()).buildClient()
+
+        when:
+        cpkncesContainer.createWithResponse(null, null, null, null)
+
+        then:
+        def e = thrown(IllegalArgumentException)
+        e.getMessage() == "x-ms-deny-encryption-scope-override is not supported for create container in service version 2019-02-02"
```

Comment:
I think testing against the literal value of the message is brittle; checking that an IllegalArgumentException is thrown is probably sufficient. 

Also, perhaps IllegalStateException is more appropriate? I think of IlegalArgumentExceptions as resulting from parameter validation on method call time.

Reply:
I'll change it to IllegalStateException. That makes sense.

Reply:
Though I think it might be best to keep the messages for now - in case we need to change the test later we can. I dont want the wrong message to be called either.